test/property_test.c

```diff
@@ -383,6 +384,34 @@ static int test_query_cache_stochastic(void)
     return res;
 }
 
+static int test_fips_mode(void)
+{
+    int ret = 0;
+    OPENSSL_CTX *ctx = NULL;
+
+    if (!TEST_ptr(ctx = OPENSSL_CTX_new()))
+        goto err;
+
+    ret = TEST_true(EVP_set_default_properties(ctx, "default=yes,fips=yes"))
+          && TEST_int_eq(EVP_default_properties_is_fips_enabled(ctx), 1)
+          && TEST_true(EVP_set_default_properties(ctx, "fips=no,default=yes"))
+          && TEST_int_eq(EVP_default_properties_is_fips_enabled(ctx), 0)
+          && TEST_true(EVP_set_default_properties(ctx, "fips=no"))
+          && TEST_int_eq(EVP_default_properties_is_fips_enabled(ctx), 0)
+          && TEST_true(EVP_set_default_properties(ctx, "fips!=no"))
+          && TEST_int_eq(EVP_default_properties_is_fips_enabled(ctx), 1)
+          && TEST_true(EVP_set_default_properties(ctx, "fips=no"))
+          && TEST_int_eq(EVP_default_properties_is_fips_enabled(ctx), 0)
+          && TEST_true(EVP_set_default_properties(ctx, "fips=no,default=yes"))
+          && TEST_true(EVP_default_properties_enable_fips(ctx, 1)
+          && TEST_int_eq(EVP_default_properties_is_fips_enabled(ctx), 1)
+          && TEST_true(EVP_default_properties_enable_fips(ctx, 0))
+          && TEST_int_eq(EVP_default_properties_is_fips_enabled(ctx), 0));
+err:
+    return ret;
```

Comment:
@paulidale - you were so busy niting me that you didnt see the memory leak :)

Reply:
Meh, it's a test and automated tools are better at picking these up.

PS: they weren't nits for the most part but logic problems :)